pkg/api/norman/customization/gke/listers.go

```diff
@@ -0,0 +1,185 @@
+package gke
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rancher/norman/httperror"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/container/v1"
+	gkeapi "google.golang.org/api/container/v1"
+	"google.golang.org/api/iam/v1"
+	"google.golang.org/api/option"
+)
+
+func getTokenSource(ctx context.Context, credential string) (oauth2.TokenSource, error) {
+	ts, err := google.CredentialsFromJSON(ctx, []byte(credential), gkeapi.CloudPlatformScope)
+	if err != nil {
+		return nil, err
+	}
+	return ts.TokenSource, nil
+}
+
+func getComputeServiceClient(ctx context.Context, credentialContent string) (*compute.Service, error) {
+	ts, err := getTokenSource(ctx, credentialContent)
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := compute.NewService(ctx, option.WithHTTPClient(oauth2.NewClient(ctx, ts)))
+	if err != nil {
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func getIamServiceClient(ctx context.Context, credentialContent string) (*iam.Service, error) {
+	ts, err := getTokenSource(ctx, credentialContent)
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := iam.NewService(ctx, option.WithHTTPClient(oauth2.NewClient(ctx, ts)))
+	if err != nil {
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func getContainerServiceClient(ctx context.Context, credentialContent string) (*container.Service, error) {
+	ts, err := getTokenSource(ctx, credentialContent)
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := container.NewService(ctx, option.WithHTTPClient(oauth2.NewClient(ctx, ts)))
+	if err != nil {
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func listMachineTypes(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	if cap.ProjectID == "" || cap.Zone == "" {
+		return nil, httperror.InvalidOption.Status, fmt.Errorf("projectId and zone are required")
```

Comment:
This is not same status code as before. Need to either confirm they are different with Wes or keep the same.

Reply:
@westlywright can you please judge on this ? do you know if HTTP error CODE is checked somewhere in UI? As I have checked and tested it is not, but please confirm. According to comment from @cmurphy we would like to use more proper HTTP error codes #31759 (comment)

Reply:
I have added back old error code response